### ISSUE TITLE
Doc: how to exclude tests in Maven build

### ISSUE THREAD

**Issue by elharo** [uber/NullAway#162]
For gradle the docs have:

```
tasks.withType(JavaCompile) {
  // remove the if condition if you want to run NullAway on test code
  if (!name.toLowerCase().contains("test")) {
    options.compilerArgs += ["-Xep:NullAway:ERROR", "-XepOpt:NullAway:AnnotatedPackages=com.uber"]
  }
}
```

It's not immediately obvious how to do this in a maven project. Please expand the maven docs to demonstrate.

**Comment by jeandersonbc** [comment]
If you want to skip tests, I believe that adding -DskipTests will bypass the Maven surefire plugin.
I'm not sure about the "verify" step, though.

More info:
* https://maven.apache.org/surefire/maven-surefire-plugin/examples/skipping-tests.html
* https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#skipTests

**Comment by elharo** [comment]
The question is not how to skip running the tests. Rather it is how to keep NullAway from checking the tests for null problems.

**Comment by jeandersonbc** [comment]
Oh, sorry for the misunderstood.

**Comment by msridhar** [comment]
I'm not a Maven expert. @kageiit any ideas or thoughts on who would know
this? Basically need a way to tweak the "main" javac args but not those for
test code.

**Comment by kageiit** [comment]
Seems like the maven compiler plugin allows configuration of main and test compiler arguments separately

http://maven.apache.org/plugins-archives/maven-compiler-plugin-2.5.1/compile-mojo.html#compilerArguments

http://maven.apache.org/plugins-archives/maven-compiler-plugin-2.5.1/testCompile-mojo.html#testCompilerArguments

**Comment by msridhar** [comment]
Thanks @kageiit.  We actually need the `compilerArgs` parameter from more recent version to pass strings:

https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#compilerArgs

https://maven.apache.org/plugins/maven-compiler-plugin/testCompile-mojo.html#compilerArgs

Our [Maven example](https://github.com/uber/NullAway/wiki/Configuration#maven) currently sets `compilerArgs` in the global configuration, which I think will apply to both the `compile` and `testCompile` phases.  Does someone know of an example somewhere that shows how to set `compilerArgs` differently for these two phases?  If someone gets me an example I can update the docs.

**Comment by cbruegg** [comment]
There actually seems to be no need to configure it for the two phases separately. I've had success just excluding the test sources path:

```xml
<compilerArgs>
  <arg>-Xep:NullAway:ERROR</arg>
  <arg>-XepOpt:NullAway:AnnotatedPackages=[...]</arg>
  <arg>-XepExcludedPaths:.*/src/test/java/.*</arg>
</compilerArgs>
```

**Comment by msridhar** [comment]
@cbruegg yes that will work if you are using the default filesystem layout.  And it could be adapted if you're doing something else.  I still suspect there is some configuration magic for passing different compiler args for test vs. main code, but we are not Maven experts and haven't figured it out.
